### ISSUE TITLE
KWDual outputs a negative mixture weight

### ISSUE THREAD

**Issue by pcarbo** [stephens999/ashr#55]
Here is the example that produces a negative mixture weight with `optmethod = mixIP` only:

```
library(ashr)
betahat <- c(-5.8336751,1.0585307,0.8701328,0.9644079,-1.0796877,-7.7252394,-2.0433250,42.5216280,3.4243654)
sebeta  <- c(0.5003549,0.4223690,0.3462631,0.3316413,0.4076427,0.5276066,0.4302293,0.2051576,0.4270015)
out1    <- ash(betahat,sebeta,method = "fdr",mixcompdist = "normal",optmethod = "mixEM")
out2    <- ash(betahat,sebeta,method = "fdr",mixcompdist = "normal",optmethod = "mixIP")
```

The *ad hoc* solution right now is to set all negative mixture weights to 0 (or a small value near 0), and re-normalize the mixture weights. It would be good to come up with a better solution at some point.

Thanks to Wei for identifying this issue.

**Comment by gaow** [comment]
I think the least thing we can do, in addition to the proposed ad hoc solution, is to capture this line of warning and convert it to something more informative:

```
Warning in REBayes::KWDual(A, rep(1, k), normalize(w), control = control) :
  estimated mixing distribution has some negative values:
		       consider reducing rtol
```

Because when using `ashr` as a module (for FLASH or in DSC) it is difficult to track these weights. A message telling users that the negative values are generated yet taken care of would be reassuring.

**Comment by pcarbo** [comment]
@gaow Do you have a small example using simulated data in which REBayes yields negative mixture weights? I'd like to add this as an example so we can use it later for testing.

**Comment by gaow** [comment]
Ahh it is complicated ... I got this warning from FLASH on 300X300 matrix. But I can now upgrade my `ashr` and rerun and see if it works.

**Comment by pcarbo** [comment]
@gaow Yes, all we need is one example from running FLASH; since FLASH runs ashr multiple times, just pick one of the iterations.

**Comment by gaow** [comment]
Ok @pcarbo then here you go:

[ash-issue55.tar.gz](https://github.com/stephens999/ashr/files/1209188/ash-issue55.tar.gz)

**Comment by gaow** [comment]
It works -- the new warning shows up, though the old warning is not masked (not a big deal). Thanks!

**Comment by pcarbo** [comment]
@gaow Here's a simple reproducible example from the script you shared with me:

```
betahat <- c(8.115,9.027,9.289,10.097,9.463)
sebeta  <- c(0.6157,0.4129,0.3197,0.3920,0.5496)
ash(betahat,sebeta,mixcompdist = "normal")
```
